docs/sources/configuration/_index.md

```diff
@@ -260,7 +260,10 @@ The `querier_config` block configures the Loki Querier.
 # CLI flag: -querier.query-ingesters-within
 [query_ingesters_within: <duration> | default = 0s]
 
-# The maximum number of concurrent queries allowed.
+# The maximum number of top-level LogQL queries that will execute at the same time, per querier process.
+# If using the query frontend, this should be set to at least (`-querier.worker-parallelism` * `number of query frontend replicas`).
+# Otherwise queries may queue in the queriers and not the frontend, which will affect QoS.
```

Comment:
This isn't quite correct. Really it should be set to match the number of CPU cores available on the querier (or slightly less than the number of cores if you are in a shared environment like we are)

For example: running one querier with 40 worker-parallelism won't gain you much if you only have 4 cores

Ideally it should be set to match the machine cores and then you should run enough queriers to match (worker-parallelism * frontend replicas)/max-concurrent

Reply:
@slim-bean Updated the doc!